### ISSUE TITLE
Flutter Web with CanvasKit: TextField with custom font won't display some characters

### ISSUE THREAD

**Issue by vs-krasnov** [flutter/flutter#58506]
There are multiple issues related to incorrect characters rendering in CanvasKit. 
I suppose this is a slightly different case because it is related to a custom font from `google_fonts`.

If I use the default font, Latin and Cyrillic characters are displayed correctly, but special characters are broken.
However, If I set a custom font (Poppins), it breaks Cyrillic characters too.

## Steps to Reproduce

1. Add the following `TextFormField` (or `TextField`) widget:
```
TextFormField(
  initialValue: "Some text, ⌘, ⌫, кириллица",
  style: GoogleFonts.poppins(),
)
```
2. Run `flutter run -d chrome --release --dart-define=FLUTTER_WEB_USE_SKIA=true`

**Expected results (DomCanvas):** 
![Flutter Demo - Google Chrome 2020-06-02 20 02 07](https://user-images.githubusercontent.com/13119003/83548232-1d6b8e00-a50c-11ea-9601-bae326301b6c.png)

**Actual results: (CanvasKit)** 

- Without custom font:
![Flutter Demo - Google Chrome 2020-06-02 20 00 38](https://user-images.githubusercontent.com/13119003/83548366-57d52b00-a50c-11ea-82e8-09c4808415a1.png)

- With custom font:
![Flutter Demo - Google Chrome 2020-06-02 19 59 29](https://user-images.githubusercontent.com/13119003/83548344-4db32c80-a50c-11ea-9a0e-04d851fe647a.png)

<details>
  <summary>Logs</summary>

```
[√] Flutter (Channel beta, 1.18.0-11.1.pre, on Microsoft Windows [Version 10.0.18363.836], locale ru-RU)
    • Flutter version 1.18.0-11.1.pre at c:\sdk\flutter
    • Framework revision 2738a1148b (3 weeks ago), 2020-05-13 15:24:36 -0700
    • Engine revision ef9215ceb2
    • Dart version 2.9.0 (build 2.9.0-8.2.beta)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\vs_kr\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit visit
      https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 46.0.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] VS Code (version 1.45.1)
    • VS Code at C:\Users\vs_kr\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.11.0

[√] Connected device (2 available)
    • Web Server • web-server • web-javascript • Flutter Tools
    • Chrome     • chrome     • web-javascript • Google Chrome 81.0.4044.138

! Doctor found issues in 1 category.
```

</details>

**Comment by TahaTesser** [comment]
<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.19.0-3.0.pre, on Mac OS X 10.15.5 19F96, locale
    en-GB)
    • Flutter version 1.19.0-3.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision 6135091de9 (31 hours ago), 2020-06-01 17:17:03 -0700
    • Engine revision e39301f23f
    • Dart version 2.9.0 (build 2.9.0-11.0.dev 6489a0c68d)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (4 available)
    • iPhone 11  • 0EC80516-8F99-4A0C-89ED-2273579862F9 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
    • macOS      • macOS                                • darwin-x64     • Mac
      OS X 10.15.5 19F96
    • Web Server • web-server                           • web-javascript •
      Flutter Tools
    • Chrome     • chrome                               • web-javascript •
      Google Chrome 83.0.4103.61

• No issues found!
``` 

</details>

**Comment by wilburt** [comment]
Happening on my end too. The characters '@' and '*' doesn't render on the web with some fonts when Skia is enabled.

**Comment by sunjianan9900** [comment]
@hterkelsen 
thank you for fix this issuse
it works and show chinese characters, but there are some Performance issue

run with --dart-define=FLUTTER_WEB_USE_SKIA=true
<img width="493" alt="skia2" src="https://user-images.githubusercontent.com/10936316/103328757-09354280-4a95-11eb-86ff-2ba4266cfbd2.png">

without --dart-define=FLUTTER_WEB_USE_SKIA=true
<img width="480" alt="noskia" src="https://user-images.githubusercontent.com/10936316/103328765-0f2b2380-4a95-11eb-9286-cf23608cc8b5.png">

and on release version it shows blank screen

**Comment by yjbanov** [comment]
Reopening as https://github.com/flutter/engine/pull/23096 was reverted

**Comment by pcsosinski** [comment]
@yjbanov please update to an appropriate milestone

**Comment by rodion-m** [comment]
Hello! Is there any forecast when this fix will be available on `flutter master`?

**Comment by joaotbelo** [comment]
@hterkelsen I noticed that when Noto fonts are used as fallback and downloaded, it blocks the UI completely.

I was trying to avoid this by including the relevant Noto fonts in the package (Noto Sans Symbols and Noto Color Emoji), but even with a simple emoji such as "🇺🇸" it ends up downloading the fonts from the web. Is there a specific font family name that needs to be used? I've tried a few variations but it didn't work.

I used the same approach with MaterialIcons and worked well.

Extract of FontManifest.json:

[
  {
    "family": "MaterialIcons",
    "fonts": [
      {
        "asset": "fonts/MaterialIcons-Regular.otf"
      }
    ]
  },
  {
    "family": "NotoColorEmojiCompat",
    "fonts": [
      {
        "asset": "fonts/noto_color_emoji.ttf"
      }
    ]
  },
  {
    "family": "NotoSansSymbols",
    "fonts": [
      {
        "asset": "fonts/noto_sans_symbols_regular.ttf"
      }
    ]
  }

Thanks for your help!
